packages/dashboard/src/components/widgets/primitives/input/index.tsx

```diff
@@ -1,17 +1,119 @@
 import React, { useState, useEffect } from 'react';
 import { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
-import { Button, Select, SelectProps } from '@cloudscape-design/components';
-import SpaceBetween from '@cloudscape-design/components/space-between';
+import { Button, Select, SelectProps, ButtonProps, StatusIndicator, Modal, Box } from '@cloudscape-design/components';
+import { ErrorDetails } from '@aws-sdk/client-iotsitewise';
 import { InputWidget as InputWidgetType } from '~/types';
+import useUpdateAssetPropertyValue from './useUpdateAssetPropertyValue';
+import useDescribeAssetProperty from './useDescribeAssetProperty';
+import useWritableSiteWiseProperty from './useWritableSiteWiseProperty';
+import './index.css';
 
 export type InputWidgetProps = InputWidgetType & { readOnly: boolean };
 
 const Input: React.FC<InputWidgetProps> = ({ readOnly, ...widget }) => {
-  const options = widget.options.map(({ label }) => ({ label, value: label }));
+  const [showConfirmationModal, setShowConfirmationModal] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [infoMessage, setInfoMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [isSending, setIsSending] = useState<boolean>(false);
+  const options = widget.properties.options.map(({ label }) => ({ label, value: label }));
   const [selectedOption, setSelectedOption] = useState<SelectProps.Option | null>(options[0]);
-  const disabled = !readOnly || options.length === 0;
+  const showSubmitAndOptions = options.length > 0 && !infoMessage && widget.assets?.[0]?.properties?.length;
   const isSelectedOptionValid = options.filter(({ label }) => selectedOption?.label === label).length > 0;
 
+  // SiteWise specific - in future we might want to add support for other data sources
+  const [propertyName, setPropertyName] = useState<string>();
+  const updateAssetPropertyValue = useUpdateAssetPropertyValue();
+  const describeAssetProperty = useDescribeAssetProperty();
+  const isWritable = useWritableSiteWiseProperty();
+
+  // For now we will only support one SiteWise Asset Property
```

Comment:
one of the great things about hooks is you don't need to write them all inlined in your component in the same way that you would for class based components (due to needing to have all the lifecycle methods contained in the class itself). can greatly increase readability by separating out the hook, to give more seperation between the visual component, and the hundred line hooks

Reply:
Definitely.